src/components/oculus-touch-controls.js

```diff
@@ -0,0 +1,199 @@
+var registerComponent = require('../core/component').registerComponent;
+var bind = require('../utils/bind');
+
+// FIXME: need appropriate models, not the vive ones!
+var TOUCH_CONTROLLER_MODEL_OBJ_URL = 'https://cdn.aframe.io/controllers/vive/vr_controller_vive.obj';
+var TOUCH_CONTROLLER_MODEL_OBJ_MTL = 'https://cdn.aframe.io/controllers/vive/vr_controller_vive.mtl';
+
+/**
+ * Oculus Touch Controls Component
+ * Interfaces with Oculus Touch controllers and maps Gamepad events to
+ * common controller buttons: trackpad, trigger, grip, menu and system
+ * It loads a controller model and highlights the pressed buttons
+ */
+module.exports.Component = registerComponent('oculus-touch-controls', {
+  dependencies: ['tracked-controls'],
+
+  schema: {
+    idPrefix: { default: 'Oculus Touch' },
+    hand: {default: 'left'},
+    buttonColor: { default: '#FAFAFA' },  // Off-white.
+    buttonHighlightColor: {default: '#22D1EE'},  // Light blue.
+    model: {default: true},
+    rotationOffset: {default: -999} // use -999 as sentinel value to auto-determine based on hand
+  },
+
+  // buttonId
+  // 0 - trackpad
+  // 1 - trigger ( intensity value from 0.5 to 1 )
+  // 2 - grip
+  // 3 - menu ( dispatch but better for menu options )
+  // 4 - system ( never dispatched on this layer )
+  mapping: {
```

Comment:
We should map the buttons to the actual names / labels on the oculus touch controller. Touch does not have a trackpad, grip, menu or system buttons. Oculus Touch has instead two triggers and an analog joystick. I would expect to see  `triggerIndex`, `A`, `B` ... in the mappings